src/main/java/com/adaptris/core/jcsmp/solace/SolaceJcsmpAbstractConsumer.java

```diff
@@ -175,30 +175,31 @@ public void onException(JCSMPException exception) {
   public void onReceive(BytesXMLMessage message) {
     try {
       AdaptrisMessage adaptrisMessage = getMessageTranslator().translate(message);
-      
+
       Consumer<AdaptrisMessage> successCallback = adpMessage -> {
         if(acknowledgeMode().equals(ackMode.CLIENT)) {
           getSessionHelper().commit(message);
         }
       };
-      
+
       Consumer<AdaptrisMessage> failureCallback = adpMessage -> {
-        log.error("Message failed.  Will not acknowledge.", adaptrisMessage.getObjectHeaders().get(CoreConstants.OBJ_METADATA_EXCEPTION_CAUSE));
+        log.error("Message failed.  Will not acknowledge. {}",
+            adaptrisMessage.getObjectHeaders().get(CoreConstants.OBJ_METADATA_EXCEPTION_CAUSE));

```

Comment:
Could argue that we should cast the object to an Exception, which means we would get the full stack trace rather than just `e.toString()`. What actually happens at runtime with or without the `{}` since we know that slf4j _knows and understands exceptions_

Reply:
Nothing wrong with the change itself, since we have failed, we're not going to ack which means that it'll be re-delivered? so do we want to have continual stack traces. Just an observational comment about context.

Reply:
I did check what adaptrisMessage.getObjectHeaders().get(CoreConstants.OBJ_METADATA_EXCEPTION_CAUSE) is and it seems to not be the exception but the service name, so I thought using it as a string agument is better.
Unless that was meant to use CoreConstants.OBJ_METADATA_EXCEPTION

Reply:
Ah, I assumed it was the exception, since it's not, you're bang on the money with the change; might be worth changing the text then to be `Message failed. Will not acknowledge; failure-location[{}]`

Reply:
This is all @aaron-mcgrath-adp  -> so perhaps he can say which it is meant to be...